su1/concurrent-write.go

```diff
@@ -0,0 +1,68 @@
+package su1
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+)
+
+func CreateFile(fileName string) *os.File {
+	file, err := os.Create(fileName)
+
+	if err != nil {
+		fmt.Println("Error writing to file:", err)
+		return nil
+	}
+
+	fmt.Println("File created successfully")
+
+	return file
+}
+
+func WriteWithOffset(file *os.File, index int, chunk string, mu *sync.Mutex) {
+	offset := int64(index * len(chunk))
+	mu.Lock()
+	defer mu.Unlock()
+
+	_, err := file.WriteAt([]byte(chunk+"\n"), offset)
```

Comment:
It seems like we don't need `\n` char here. Because of this here's what can happen:
<img width="699" alt="Screenshot 2024-11-06 at 21 26 35" src="https://github.com/user-attachments/assets/32f06c00-7808-45ea-8819-673567eb64a4">

Also, remove mutex and try running `go run -race .`. Even if there would be an error regarding data race the usage of mutex in this case completely negates the purpose of worker pool since all tasks are executed one by one, rather than concurrent.